arduino/scout/NewTone.cpp

```diff
@@ -0,0 +1,56 @@
+#include "NewTone.h"
+#include "KeyBuffer.h"
+#include "Arduino.h"
+
+uint16_t speaker_preload[BUFFER_MAX] = {0};
+
+// Note: rolling this into a loop causes us to miss the timing.
```

Comment:
It's baffling to me that the default optimization doesn't unroll this when I put it in a loop!

Reply:
This interrupt routine has to complete within 256 clock cycles. I haven't done any analysis as to _why_, but if I put this in a for loop and do it four times, something about how the assembly is generated makes it take too long. If I were working, that would be the next step. But I'm playing, so I just pushed the code as it worked and left a note ;-)